### PR TITLE
Prefab | Fix crash on escaping prefab edit mode

### PR DESCRIPTION
**Pull request by forkercat**
Signed-off-by: Junhao Wang <wjunhao@amazon.com>

## What does this PR do?

Fix: #11741

This PR fixes the crash when escaping prefab edit mode. See the issue for more details.

The changes were originally introduced from prefab overrides UX prototype #10943. The related two lines should be put under the overrides UX flag.

**Note: The flag is OFF by default and the two lines were added after the above PR. So the risk of merging this PR should be minimal.**

## How was this PR tested?

- [x] Manually tested in development. The crash does not occur any more when the flag is off or on.
- [x] Manually tested in stabilization/2210 branch

### CODE REVIEW

**Review by michabr — APPROVED** [review]

**Comment by tonybalandiuk** [comment]
This PR needs to be submitted via the exception process since we are now in code freeze.
 I am adding it to the exception board for tracking (step 1). @monroegm @lsemp3d  Can you see to Step 2 and 3. 
@bhanuja-s   @rainbj for Step 4 
 @AMZN-Dk  @Kadino  for Step 5

 **Exception Tracking**

* [X] 1. Owning SIG requests the exception, by a Sig Representative (Chair/co-chair) adding the PR to the exception requests queue board.
* [ ] 2. Owning SIG comments on the PR explaining impact if this code does not go into stabilization.
* [ ] 3. Unrelated SIG representative (chair/cochair) approves the exception, adding a comment indicating approval and the SIG they represent.
* [ ] 4. UX SIG representative (chair/cochair) approves the exception, adding a comment indicating approval and the SIG they represent.
* [ ] 5. Testing SIG representative (chair/cochair) approves the exception, adding a comment indicating approval and the SIG they represent.
* [ ] (Must occur at some point before SIG-Release approval) Code has been approved for merging by maintainers
* [ ] 6. SIG-Release Release Manager or co-Release Manager approves the exception, adding a comment indicating approval and the SIG they represent.
* [ ] 7. If all approvals are accepted, then the exception PR may be merged
* [ ] 8. Merged PR is verified successful via testing. If Merged PR verification testing failed, consult reps from step 3-5 and discussion occurs on fixing the failure vs backing out change
* [ ] 9. Exception is complete

**Comment by forkercat** [comment]
**Exception Tracking - Step 2:** If this PR does not go into stabilization, this crash would occur when users try to exit prefab edit mode for this particular nested prefab hierarchy. The crash usually occurs at/after the second time of exiting.

**Review by srikappa-amzn — APPROVED** [review]
Looks good for a stabilization fix. 

@forkercat , can you turn on overrides ux flag and see if this reproduces in override mode? If it does, we need to find a solution for that too.

**Comment by forkercat** [comment]
> @forkercat , can you turn on overrides ux flag and see if this reproduces in override mode? If it does, we need to find a solution for that too.

Yeah. I verified that the crash didn't occur in stabilization when the flag is on.

**Comment by rainbj** [comment]
@srikappa-amzn Can we do a quick video chat, so I can see the workflow in action. Then I can sign off on this ticket. Thanks

**Comment by forkercat** [comment]
> Can we do a quick video chat, so I can see the workflow in action. Then I can sign off on this ticket. Thanks

@rainbj I put up a recording to demonstrate what the workflow and hierarchy is. Let me know if you have any question.

https://user-images.githubusercontent.com/11157226/193655567-64d20c12-548a-4aa7-8273-8a2f08e409bc.mp4

**Comment by forkercat** [comment]
Tagging @AMZN-Dk @Kadino for @o3de/sig-testing approval. Thanks!

**Comment by forkercat** [comment]
Requesting exemption from @o3de/sig-graphics-audio for step 3.

**Comment by rainbj** [comment]
Sig-UI-UX approved.

**Comment by AMZN-Dk** [comment]
Agreed this should be fixed for the release, bad issue for prefab UX, approved from Testing SIG

**Comment by tonybalandiuk** [comment]
approved as release manager on behalf of sig-release. This still requires secondary-sig approval before it can be merged. Please reach out to another SIG to get their approval. thanks

**Comment by forkercat** [comment]
Got another testing from @dm-amzn and verified that the crash is fixed in stabilization on this branch.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp
Comment (lumberyard-employee-dm) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp:337]: Why was the openState parameter changed from `m_prefabEditScope == PrefabEditScope::SHOW_NESTED_INSTANCES_CONTENT` -> `true`?

Comment (forkercat) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp:337]: I notice the condition always returns True when overrides ux flag is enabled, so I explicitly change it to True. And even I change it to True, I am still seeing that the nested container entities are not opened correctly. This could be related to some outliner collapsing issues we discussed recently.

Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp:337]:
```
m_prefabEditScope == PrefabEditScope::SHOW_NESTED_INSTANCES_CONTENT
```

This was added as future-proofing, but it is indeed hardcoded to always be true. If possible, I'd revert it to the check as that is meant to enable switching back and forth between overrides UX and previous UX at runtime.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Comment by lumberyard-employee-dm** [comment]
Approving the exception as an representative of SIG-Core as an unrelated SIG

**Comment by tonybalandiuk** [comment]
This can be merged @forkercat - it has received all approvals.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp
Comment (forkercat) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp:337]: Discussed offline with @AMZN-daimini. This condition can be kept for switching edit scope at runtime, which is useful for future use case on pinning property in DPE. For this into stab, it is okay to have it be deterministic to return True if flag is on.

We agreed that we should bring back this condition into dev in a separate PR so this one can be unblocked.

**Pull request merged**
